--- include/linux/if_bridge.h	2016-08-08 23:58:39.000000000 +0800
+++ include/linux/if_bridge.h	2016-08-24 11:06:38.158485106 +0800
@@ -15,7 +15,6 @@
 
 #include <linux/types.h>
 #include <linux/if_ether.h>
-#include <linux/in6.h>
 
 #define SYSFS_BRIDGE_ATTR	"bridge"
 #define SYSFS_BRIDGE_FDB	"brforward"
--- include/linux/xfrm.h	2016-08-08 23:58:39.000000000 +0800
+++ include/linux/xfrm.h	2016-08-24 11:03:47.238489586 +0800
@@ -1,7 +1,6 @@
 #ifndef _LINUX_XFRM_H
 #define _LINUX_XFRM_H
 
-#include <linux/in6.h>
 #include <linux/types.h>
 
 /* All of the structures in this file may not change size as they are
--- include/utils.h	2016-08-08 23:58:39.000000000 +0800
+++ include/utils.h	2016-08-24 11:04:41.642488160 +0800
@@ -209,7 +209,9 @@
 
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
 
+#ifndef BUILD_BUG_ON
 #define BUILD_BUG_ON(cond) ((void)sizeof(char[1 - 2 * !!(cond)]))
+#endif
 
 #ifndef offsetof
 # define offsetof(type, member) ((size_t) &((type *)0)->member)
--- Makefile	2016-08-08 23:58:39.000000000 +0800
+++ Makefile	2016-08-24 11:13:25.938474417 +0800
@@ -4,7 +4,7 @@
 
 PREFIX?=/usr
 LIBDIR?=$(PREFIX)/lib
-SBINDIR?=/sbin
+SBINDIR?=$(PREFIX)/sbin
 CONFDIR?=/etc/iproute2
 DATADIR?=$(PREFIX)/share
 DOCDIR?=$(DATADIR)/doc/iproute2
@@ -46,7 +46,7 @@
 CFLAGS := $(WFLAGS) $(CCOPTS) -I../include $(DEFINES) $(CFLAGS)
 YACCFLAGS = -d -t -v
 
-SUBDIRS=lib ip tc bridge misc netem genl tipc devlink man
+SUBDIRS=lib ip
 
 LIBNETLINK=../lib/libnetlink.a ../lib/libutil.a
 LDLIBS += $(LIBNETLINK)
