--- Makefile	2016-08-18 20:36:51.911212776 +0800
+++ Makefile	2016-08-24 15:23:44.902080727 +0800
@@ -18,13 +18,13 @@
 
 # Variables you may well want to override.
 
-PREFIX        = /usr/local
+PREFIX        ?= /usr/local
 BINDIR        = $(PREFIX)/sbin
 MANDIR        = $(PREFIX)/share/man
 LOCALEDIR     = $(PREFIX)/share/locale
 BUILDDIR      = $(SRC)
-DESTDIR       = 
-CFLAGS        = -Wall -W -O2
+DESTDIR       ?= 
+CFLAGS        += -Wall -W -O2
 LDFLAGS       = 
 COPTS         = 
 RPM_OPT_FLAGS = 
