--- a/Makefile	2014-03-22 03:26:36.000000000 +0800
+++ b/Makefile	2016-08-26 20:28:28.446878492 +0800
@@ -1,7 +1,7 @@
 # Makefile for linuxtv.org dvb-apps
 
 # get DVB API version
-VERSION_FILE := "/usr/include/linux/dvb/version.h"
+VERSION_FILE ?= "/usr/include/linux/dvb/version.h"
 
 DVB_API_MAJOR := $(word 3, $(shell grep -m1 "DVB_API_VERSION" $(VERSION_FILE)) )
 DVB_API_MINOR := $(word 3, $(shell grep -m1 "DVB_API_VERSION_MINOR" $(VERSION_FILE)) )
@@ -10,8 +10,6 @@
 
 all clean install:
 	$(MAKE) -C lib $@
-	$(MAKE) -C test $@
-	$(MAKE) -C util $@
 
 update:
 	@echo "Pulling changes & updating from master repository"
