diff --git a/make-linux.mk b/make-linux.mk
index 7c77f58..bae2fb5 100644
--- a/make-linux.mk
+++ b/make-linux.mk
@@ -116,13 +116,6 @@ ifeq ($(CC_MACH),aarch64)
 endif
 DEFS+=-DZT_BUILD_PLATFORM=1 -DZT_BUILD_ARCHITECTURE=$(ZT_ARCHITECTURE) -DZT_SOFTWARE_UPDATE_DEFAULT="\"disable\""
 
-# Define some conservative CPU instruction set flags for arm32 since there's a ton of variation out there
-ifeq ($(ZT_ARCHITECTURE),3)
-				override CFLAGS+=-march=armv6zk -mcpu=arm1176jzf-s -mfloat-abi=hard -mfpu=vfp
-				override CXXFLAGS+=-march=armv6zk -mcpu=arm1176jzf-s -mfloat-abi=hard -mfpu=vfp
-				override DEFS+=-DZT_NO_TYPE_PUNNING
-endif
-
 # Define this to build a static binary, which is needed to make this runnable on a few ancient Linux distros
 ifeq ($(ZT_STATIC),1)
 	override LDFLAGS+=-static
